packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,341 @@ import {
     } else {
-      this.isNavigationIdle = false;
+      internalState.isNavigationIdle = false;
     }
 
-    this.navigationIdleTimeoutId = this.async.setTimeout(() => {
-      this.isNavigationIdle = true;
-    }, this.navigationIdleDelay);
-  };
+    internalState.navigationIdleTimeoutId = setTimeout(() => {
+      internalState.isNavigationIdle = true;
+    }, internalState.navigationIdleDelay);
+  }, [internalState, clearTimeout, setTimeout]);
 
   /**
-   * Callback passed to the GridCell class that will trigger the onCellHovered callback of the SwatchColorPicker
-   * NOTE: This will not be triggered if shouldFocusOnHover === true
+   * Callback to make sure we don't update the hovered element during mouse wheel
    */
-  private _onGridCellHovered = (item?: IColorCellProps): void => {
-    const { onCellHovered } = this.props;
-
-    if (onCellHovered) {
-      return item ? onCellHovered(item.id, item.color) : onCellHovered();
-    }
-  };
+  const onWheel = React.useCallback((): void => {
+    setNavigationTimeout();
+  }, [setNavigationTimeout]);
 
   /**
-   * Callback passed to the GridCell class that will trigger the onCellFocus callback of the SwatchColorPicker
+   * Callback used to handle KeyCode events
    */
-  private _onGridCellFocused = (item?: IColorCellProps): void => {
-    const { onCellFocused } = this.props;
-    if (onCellFocused) {
-      if (item) {
-        this._cellFocused = true;
-        return onCellFocused(item.id, item.color);
-      } else {
-        this._cellFocused = false;
-        return onCellFocused();
+  const onKeyDown = React.useCallback(
+    (ev: React.KeyboardEvent<HTMLButtonElement>): void => {
+      if (
+        ev.which === KeyCodes.up ||
+        ev.which === KeyCodes.down ||
+        ev.which === KeyCodes.left ||
+        ev.which === KeyCodes.right
+      ) {
+        setNavigationTimeout();
       }
-    }
-  };
+    },
+    [setNavigationTimeout],
+  );
 
   /**
-   * Handle the click on a cell
-   * @param item - The cell that the click was fired against
+   * Render a color cell
+   * @param item - The item to render
+   * @returns - Element representing the item
    */
-  private _onCellClick = (item: IColorCellProps): void => {
-    if (this.props.disabled) {
-      return;
-    }
-
-    const index = item.index as number;
 
-    // If we have a valid index and it is not already
-    // selected, select it
-    if (index >= 0 && index !== this.state.selectedIndex) {
-      if (this.props.onCellFocused && this._cellFocused) {
-        this._cellFocused = false;
-        this.props.onCellFocused();
-      }
-
-      if (this.props.onColorChanged) {
-        this.props.onColorChanged(item.id, item.color);
-      }
+  const renderOption = React.useCallback(
+    (item: IColorCellProps): JSX.Element => {
+      return (
+        <ColorPickerGridCell
+          item={item}
+          idPrefix={id}
+          color={item.color}
+          styles={props.getColorGridCellStyles}
```

Comment:
Same about destructuring. Though in this case it might be better to just *not* use `useCallback` and let it re-generate every time, since that will likely happen anyway due to the number of deps... Also since looking at the code of Grid, nothing else is memoized based on this function.